packages/twenty-server/src/engine/core-modules/workspace/services/workspace.service.ts

```diff
@@ -61,8 +62,14 @@ export class WorkspaceService extends TypeOrmQueryService<Workspace> {
         payload.subdomain,
       );
 
-      if (!subdomainAvailable) {
-        throw new ConflictError('Subdomain already taken');
+      if (
+        !subdomainAvailable ||
```

Comment:
here we are adding the check updateWorkspaceById method which is closer to the model layer. This is fine as long as we say that we will always use this method to update a workspace and not use the repository directly. 
I feel our model layer is typeorm and an approach like this https://orkhan.gitbook.io/typeorm/docs/validation would be more robust